assets/js/admin/course-edit.js

```diff
@@ -1,4 +1,20 @@
 jQuery( document ).ready( function ( $ ) {
+	const editPostSelector = wp.data.select( 'core/edit-post' );
+	const editPostDispatcher = wp.data.dispatch( 'core/edit-post' );
+
+	const isLegacyMetaBoxesDisabled =
+		! editPostSelector.isEditorPanelEnabled( 'meta-box-course-lessons' ) ||
+		! editPostSelector.isEditorPanelEnabled( 'meta-box-course-lessons' );
+
+	if ( isLegacyMetaBoxesDisabled ) {
```

Comment:
I think these two should be checked separately, if only one of them is disabled, the toggle would enable one and disable the other.

Reply:
Makes sense! Updated here: https://github.com/Automattic/sensei/pull/3661/commits/8a2d4dcaf0f23e98ff19b1af6706752131555548